### PR TITLE
Add how to troubleshoot skipped but required checks

### LINKED ISSUE #8926: Explain how to handle skipped required checks
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's Code of Conduct.

### What article on docs.github.com is affected?

https://github.com/github/docs/blob/2289ca70c7ec7ec1b81eafdb333cd0157fafaa08/content/github/administering-a-repository/troubleshooting-required-status-checks.md

### What part(s) of the article would you like to see updated?

A section should be added that explains that if people use `paths` for a required check, they'll want to add a second copy of the check with `paths-ignore` for the same content that just returns `true`.

### Additional information

It's a FAQ in https://github.community

### PR DESCRIPTION
**Pull request by thispsj**
### Why:

Closes #8926 

### What's being changed:

This adds a troubleshooting guide as to how to handle skipped but required checks on a PR.

### Check off the following:

- [x] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@thispsj Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Comment by mostimoti** [comment]
https://github.com/github/docs/projects/1#card-70814394

**Comment by mostimoti** [comment]
https://github.com/github/docs/projects/1#card-70814394

**Comment by mostimoti** [comment]
FB_IMG_1634157941356

# ok

File: content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md [hunk 1]
```diff
@@ -38,11 +38,87 @@ remote: error: Required status check "ci-build" is failing
...
+      - 'middleware/**'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - run: 'echo "No build required" '
+```
+Now the checks will always pass whenever someone sends a pull request which doesn't change the files listed under the `paths` key of the actual workflow.
+
+![Check skipped but passes due to Generic workflow](/assets/images/help/repository/PR-required-check-passed-using-generic.jpeg)
```
Comment (hubwriter) [inline content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md:107]:
```suggestion
![Check skipped but passes due to generic workflow](/assets/images/help/repository/pr-required-check-passed-using-generic.png)
```

File: content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md [hunk 2]
```diff
@@ -38,11 +38,87 @@ remote: error: Required status check "ci-build" is failing
...
+    paths-ignore:
+      - 'scripts/**'
+      - 'middleware/**'
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - run: 'echo "No build required" '
+```
+Now the checks will always pass whenever someone sends a pull request which doesn't change the files listed under the `paths` key of the actual workflow.
```
Comment (hubwriter) [inline content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md:105]:
```suggestion
Now the checks will always pass whenever someone sends a pull request that doesn't change the files listed under `paths` in the first workflow.
```

File: content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md [hunk 3]
```diff
@@ -38,11 +38,87 @@ remote: error: Required status check "ci-build" is failing
...
+{% note %}
+
+**Note**: Make sure that the `name` key and required job name in both the workflow files are the same. For more information, see "[Workflow Syntax for GitHub Actions](/actions/reference/workflow-syntax-for-github-actions)".
+
+{% endnote %}
+
+
+{% note %}
+
+**Note**: The example above uses {% data product.prodname_actions %} but this workaround is also applicable to other CI/CD providers that integrate with {% data product.company_short %}.
```
Comment (hubwriter) [inline content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md:118]:
```suggestion
**Notes:** 

* Make sure that the `name` key and required job name in both the workflow files are the same. For more information, see "[Workflow syntax for {% data variables.product.prodname_actions %}](/actions/reference/workflow-syntax-for-github-actions)".
* The example above uses {% data variables.product.prodname_actions %} but this workaround is also applicable to other CI/CD providers that integrate with {% data variables.product.company_short %}.
```

**Review by hubwriter — COMMENTED** [review]
@thispsj - Many thanks for raising this PR. Great work. This is a really useful addition and clearly something people have struggled with, so it's great to get it documented.

I've suggested quite a few changes. These are all just stylistic things, mainly to fit with our style guide. I hope you are okay with this and can just click to commit these changes.

I'd also like to use different screenshots. Yours are good but just use a different highlighting style than we use in the rest of the docs. I'll add a couple of replacement screenshots. Could you swap these into your PR please?

Thanks again for taking the time to do this docs work. Your efforts are much appreciated. Apologies it's taken a while to get to reviewing this. 🎖️

**Comment by hubwriter** [comment]
### Replacement screenshots

Please save this as `pr-required-check-skipped.png`:

pr-required-check-skipped

Please save this as `pr-required-check-passed-using-generic.png`:

pr-required-check-passed-using-generic

**Review by mostimoti — COMMENTED** [review]
Ok

File: content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md [hunk 4]
```diff
@@ -38,11 +38,87 @@ remote: error: Required status check "ci-build" is failing
...
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
+
+    steps:
```
[the remark this answers is not in the document]
Comment (turkiszz) [reply content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md]: @

[the remark this answers is not in the document]
Comment (turkiszz) [reply content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md]: - [ ] @**[]()**

File: content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md [hunk 5]
```diff
@@ -38,11 +38,79 @@ remote: error: Required status check "ci-build" is failing
...
+![Check skipped but passes due to generic workflow](/assets/images/help/repository/PR-required-check-passed-using-generic.png)
+
+{% note %}
+
+**Notes:** 
+* Make sure that the `name` key and required job name in both the workflow files are the same. For more information, see "[Workflow syntax for {% data variables.product.prodname_actions %}](/actions/reference/workflow-syntax-for-github-actions)".
+* The example above uses {% data variables.product.prodname_actions %} but this workaround is also applicable to other CI/CD providers that integrate with {% data variables.product.company_short %}.
+
+{% endnote %}
+=======
```
Comment (hubwriter) [inline content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md:113]:
```suggestion

```

File: content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md [hunk 6]
```diff
@@ -38,11 +38,79 @@ remote: error: Required status check "ci-build" is failing
...
 It's also possible for a protected branch to require a status check from a specific {% data variables.product.prodname_github_app %}. If you see a message similar to the following, then you should verify that the check listed in the merge box was set by the expected app.
```
Comment (hubwriter) [inline content/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks.md:114]:
```suggestion
{% endnote %}

It's also possible for a protected branch to require a status check from a specific {% data variables.product.prodname_github_app %}. If you see a message similar to the following, then you should verify that the check listed in the merge box was set by the expected app.
```

**Review by hubwriter — APPROVED** [review]

**Comment by hubwriter** [comment]
@thispsj - Thanks for the changes.
Merging this now.

**Comment by hubwriter** [comment]
Now live: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks
🎉

**Comment by steve21168** [comment]
Unless I'm missing something this solution seems incomplete.  Say two files change, for example: "scripts/foo" and "README.md".  Wouldn't both actions trigger and then the required check would falsely get marked successful?

**Comment by thispsj** [comment]
> Unless I'm missing something this solution seems incomplete. Say two files change, for example: "scripts/foo" and "README.md". Wouldn't both actions trigger and then the required check would falsely get marked successful?

@steve21168 a good question indeed !
It's very simple to answer that : Both the workflows will run as required when there are changes to items in the required path and also to items out of the required path.

I have also created a practical demo for you which you can see [here](https://github.com/thispsj-lab/pr-actions/pull/14)

If you don't understand any specific thing, don't hesitate to ping me and ask it here.

**Comment by steve21168** [comment]
> > Unless I'm missing something this solution seems incomplete. Say two files change, for example: "scripts/foo" and "README.md". Wouldn't both actions trigger and then the required check would falsely get marked successful?
> 
> @steve21168 a good question indeed ! It's very simple to answer that : Both the workflows will run as required when there are changes to items in the required path and also to items out of the required path.
> 
> I have also created a practical demo for you which you can see [here](https://github.com/thispsj-lab/pr-actions/pull/14).
> 
> If you don't understand any specific thing, don't hesitate to ping me and ask it here.

Yes so in your demo both of your "ci/build" Github actions ran - in your example you can see there are two "ci/build" checks .  What you want to happen is to only have the one real "ci" run.  

The issue with both actions running is that the build which skips everything will pass very quickly and the status check will be marked a success making the PR mergable.  The real "ci/build" may still be running at this time.

**Comment by thispsj** [comment]
> > > Unless I'm missing something this solution seems incomplete. Say two files change, for example: "scripts/foo" and "README.md". Wouldn't both actions trigger and then the required check would falsely get marked successful?
> > 
> > 
> > @steve21168 a good question indeed ! It's very simple to answer that : Both the workflows will run as required when there are changes to items in the required path and also to items out of the required path.
> > I have also created a practical demo for you which you can see [here](https://github.com/thispsj-lab/pr-actions/pull/14).
> > If you don't understand any specific thing, don't hesitate to ping me and ask it here.
> 
> Yes so in your demo both of your "ci/build" Github actions ran - in your example you can see there are two "ci/build" checks . What you want to happen is to only have the one real "ci" run.
> 
> The issue with both actions running is that the build which skips everything will pass very quickly and the status check will be marked a success making the PR mergable. The real "ci/build" may still be running at this time.

@steve21168 you're partially correct.

It's true that the generic check will pass very quickly. But this will **not** make the PR mergable.
This is because both the real & generic checks run as **required**. You can see that in the screenshot below : 
Both run as required

**Comment by thispsj** [comment]
@steve21168 for your doubt I added a delay in the real build using `sleep` and then added a commit to the PR branch and now you can clearly see what I was trying to convey : 
Both checks are required

Also if you are still confused, why not open a PR on my repo and see it happen yourself.

**Comment by steve21168** [comment]
@thispsj Thank you for the time an effort you put into this.  In my scenario the required check is a dependency of an earlier job.  Until the earlier job finishes I only see the one check which is the bypassing check.  See the scenario setup here: https://github.com/thispsj-lab/pr-actions/pull/16

**Comment by thispsj** [comment]
> 
Ya it's like that on your side. On my side it show **2 workflows awaiting approval**. See :
2 WFlows

If you're still online put a like here and I will approve the workflows to run so that you can see that live.

**Comment by thispsj** [comment]
@steve21168 and the logic behind it showing one check is because of branch protection rules where I've define only one check but when you approve it two workflows will run.

**Comment by thispsj** [comment]
@steve21168 also you've changed the name of job it's not the same. So it'll not show as required.

**Comment by thispsj** [comment]
@steve21168 open another PR without changing name and you'll realise.

**Comment by steve21168** [comment]
Yes 

> @steve21168 and the logic behind it showing one check is because of branch protection rules where I've define only one check but when you approve it two workflows will run.

Yes both workflows run - but the branch protection check would be falsely marked successful b/c you have a job that just goes straight to success - defeating the purpose of the branch protection.  The second workflow may eventually mark the required check as failed but there could be a long period where it is marked successful.

@thispsj Thank you for your time.

**Comment by thispsj** [comment]
> I have not changed the name they are both "ci/build" I just put a step before to show that until it gets there the build would be mergeable since the first "ci/build" was successful and the second "ci/build" has yet to appear.
> 
> Now that both jobs have caught up you can see the two required "ci/build" checks.. But until "slow-job" finished the build had one successful "ci/build" check and was therefore mergeable.

Ahh I got it when the real build job is invoked directly (i.e. without depending on another job) this fix in GitHub Docs would work perfectly because the real build job would start along with the generic one. But when real build job is dependent on another job, the PR will show as mergable till the job on which build is dependent is completed. Perfect ! Looks like you've found something new to add to this document.

**Comment by steve21168** [comment]
@thispsj You can go for it - I'm sure the community will appreciate it.

**Comment by thispsj** [comment]
@steve21168 Thanks !
Had a great time conversing with you. Definitely improvised my knowledge about GitHub Actions.

**Comment by andthum** [comment]
> > I have not changed the name they are both "ci/build" I just put a step before to show that until it gets there the build would be mergeable since the first "ci/build" was successful and the second "ci/build" has yet to appear.
> > Now that both jobs have caught up you can see the two required "ci/build" checks.. But until "slow-job" finished the build had one successful "ci/build" check and was therefore mergeable.
> 
> @steve21168 Ahh I got it when the real build job is invoked directly (i.e. without depending on another job) this fix in GitHub Docs would work perfectly because the real build job would start along with the generic one. But when real build job is dependent on another job, the PR will show as mergable till the job on which build is dependent is completed. Perfect ! Looks like you've found something new to add to this document.

I now this thread is more than one year old, but I just stumbled over the same issue and would suggest the following solution: Couldn't you just make the dummy build-job dependent on the same job as the real build-job? Then both build-jobs (dummy and real) should again start at the same time and there shouldn't be a time window where the dummy job finished successfully while the real job still didn't started and hence there shouldn't be any time window where the PR is mergable although the real build-job hasn't started yet.

**Pull request merged**